packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -405,8 +405,15 @@ export async function updateConfigOnEnvInit(context: $TSContext, resourceName: s
       }
 
       const currentCfnTemplatePath = pathManager.getCurrentCfnTemplatePath(projectPath, categoryName, resourceName);
-      const { cfnTemplate: currentCfnTemplate } = await readCFNTemplate(currentCfnTemplatePath);
-      await writeCFNTemplate(currentCfnTemplate, pathManager.getResourceCfnTemplatePath(projectPath, categoryName, resourceName));
+      try {
+        const { cfnTemplate: currentCfnTemplate } = await readCFNTemplate(currentCfnTemplatePath);
+        await writeCFNTemplate(currentCfnTemplate, pathManager.getResourceCfnTemplatePath(projectPath, categoryName, resourceName));
+      } catch (e) {
+        if (e.message === `No CloudFormation template found at ${currentCfnTemplatePath}`) {
```

Comment:
might be better to update `readCFNTemplate` to accept a `throwIfNotExist` option rather than catch and parse the error here. This implementation tightly couples the error message to business logic (ie updating the error message would cause the bug to resurface)